### PR TITLE
feat: fiat ramp JunoPay

### PR DESCRIPTION
**Pull request by 0xdef1cafe**
## Description

requires https://github.com/shapeshift/lib/pull/863

Adds JunoPay onramp to fiat onramps.

## Notice

- [x] Have you followed the guidelines in our [Contributing]('https://github.com/shapeshift/web/CONTRIBUTING.md) guide?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/shapeshift/web/pulls) for the same update/change?

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Testing
To test, use the JunoPay fiat onramp and it should direct you to buy crypto through your Juno account. The crypto should be sent to your ShapeShift wallet.

## Screenshots (if applicable)
Screenshot 2022-07-07 at 12 17 50 AM
Screenshot 2022-07-07 at 12 18 26 AM
Screenshot 2022-07-07 at 8 18 41 PM

### CODE REVIEW

**Comment by 0xdef1cafe** [comment]
@shapeshift/operations this can be tested behind a flag in the fleek build - no need to complete a full purchase - only supports buying, just make sure the window opens correctly for ETH, ERC20s, and BTC with the correct address

**Comment by 0xdef1cafe** [comment]
> > @shapeshift/operations this can be tested behind a flag in the fleek build - no need to complete a full purchase - only supports buying, just make sure the window opens correctly for ETH, ERC20s, and BTC with the correct address
> 
> @0xdef1cafe checked [this link](https://bafybeifqsv2hrciygmwv3g7l7orjpwzoonub6n2ojblpfd4lltx3epvhcm.on.fleek.co), opens a new tab and confirmed BTC, ETH, and ERC20 address is correct.
> 
> only thing i noticed is when I clicked on "Select Coin" and even chose the same asset, the address is blank on the next page
> 
>  Screen.Recording.2022-07-09.at.10.49.46.AM.mov

thanks for testing @Lychbot! this UX is on junopay's side of things - it's up to them and probably by design. as long as the correct address and asset comes over when it opens the new tab we're 👜

**Review by gomesalexandre — APPROVED** [review]
LGTM! Tested:

### Asset selection (Tested with ETH/ERC20s and BTC)

- MetaMask, Tally and XDEFI wallets on Mainnet, correctly seeing the intersection of OnJuno/ our ETH/ERC20s as supported assets with Bitcoin disabled.
- Native, correctly seeing the same intersection, with Bitcoin enabled
- Keplr, correctly seeing all as disabled 

### Continue Modal (Tested with ETH/ERC20s and BTC)

- The address of the relevant wallet is properly populated.
- Copy button correctly copies the populated address in the system clipboard

### External URL (Tested with ETH/ERC20s and BTC)

- The ShapeShift partner URL opens populated with the right asset and wallet address

Other than the nitpicky comment, and @0xApotheosis non-blocking comments, this looks 🛍️ to me

File: src/components/Modals/FiatRamps/fiatRampProviders/junopay.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+  action: FiatRampAction,
+  asset: string,
+  address: string,
+): string => {
+  const baseUrl = new URL(getConfig().REACT_APP_JUNOPAY_BASE_APP_URL)
+  const params = new URLSearchParams()
+
+  params.set('action', action === FiatRampAction.Sell ? 'sell' : 'buy')
+  params.set('currency', asset && asset.toLowerCase())
+  params.set('partnerKey', 'live_SYkQkrlyIQeuQf8AauSOaGTa')
```
[the remark this answers is not in the document]
Comment (0xdef1cafe) [reply src/components/Modals/FiatRamps/fiatRampProviders/junopay.ts:78]: indeed we can, good catch

**Comment by 0xdef1cafe** [comment]
@0xApotheosis @gomesalexandre ty for the reviews - all implemented as suggested

**Review by gomesalexandre — APPROVED** [review]
Tested again following the previous testing criteria, LGTM 🛍️

**Comment by 0xdef1cafe** [comment]
added all flags to fleek - also verified and working in ephemeral build. merging as previously approved by @0xApotheosis and reapproved and tested by @gomesalexandre

**Pull request merged**